sql/core/src/main/scala/org/apache/spark/sql/execution/metric/SQLMetrics.scala

```diff
@@ -194,4 +202,16 @@ object SQLMetrics {
         SparkListenerDriverAccumUpdates(executionId.toLong, metrics.map(m => m.id -> m.value)))
     }
   }
+
+  /**
+   * Create all shuffle read relative metrics and return the Map.
+   */
+  def getShuffleReadMetrics(sc: SparkContext): Map[String, SQLMetric] = Map(
```

Comment:
I'd prefer to name this create, rather than get, to imply we are creating a new set rather than just returning some existing sets.

Reply:
Thanks, rename it to createShuffleReadMetrics and move to SQLShuffleMetricsReporter.